studio/src/pages/RequestorPage/reducer/reducer.ts

```diff
@@ -683,12 +705,83 @@ export function useRequestor() {
     [dispatch],
   );
 
+  /**
+   * Looks for any middleware that will match the current request
+   */
+  const getMatchingMiddleware = useCallback(() => {
+    const path = state.path;
+    const method = state.method;
+    const requestType = state.requestType;
+
+    const canMatchMiddleware =
+      !pathHasValidBaseUrl(path) || path.startsWith(state.serviceBaseUrl);
+
+    // NOTE - We can only match middleware for the service we're monitoring anyhow
+    //        If someone is making a request to jsonplaceholder, we don't wanna
+    //        match middleware that might fire for an internal goose api call
+    if (!canMatchMiddleware) {
+      return null;
+    }
+
+    const matchedRoute = findMatchedRoute(
+      state.routes,
+      removeBaseUrl(state.serviceBaseUrl, path),
+      method,
+      requestType,
+    )?.route;
+
+    if (!matchedRoute) {
+      return null;
+    }
+
+    const indexOfMatchedRoute = matchedRoute
+      ? state.routesAndMiddleware.indexOf(matchedRoute)
+      : -1;
+
+    const registeredHandlersBeforeRoute =
+      indexOfMatchedRoute > -1
+        ? state.routesAndMiddleware.slice(indexOfMatchedRoute)
+        : [];
+
+    const filteredMiddleware = registeredHandlersBeforeRoute.filter(
+      (r) => r.handlerType === "middleware",
+    );
+
+    const middlewareMatches = findAllSmartRouterMatches(
+      filteredMiddleware,
+      removeBaseUrl(state.serviceBaseUrl, path),
+      method,
+      requestType,
+    );
+
+    // console.log("matched route", matchedRoute);
```

Comment:
👋

Reply:
in the future you can add a code suggestion to remove lines like this! then i can just accept and merge ((i'll try to do the same on pr reviews))